### DISCUSSION TITLE
volar does not work

### DISCUSSION THREAD

**Discussion by KKKiraSong** [w0rp/ale#4460] [Q&A - Ask for help with problems]
I used vls for vue2, it works well.

I created a new proj with vue3 last month. So I try to use volar in ale. But I have some difficulties.

I install volar: npm i -g @volar/vue-language-server. And the version is 1.0.24.

But when I open a vue file, it doesn't work.

![image](https://user-images.githubusercontent.com/7237865/220242573-f8d7e7ce-9c75-45ed-a330-1581b3fae293.png)

I define 'activeTabs', but I use 'activeTab' in template tag. ALE cannot work.

I have installed ts in my project.

I cannot find the reason. Plz help me.

Thank u.

My ALEInfo:

![image](https://user-images.githubusercontent.com/7237865/220242820-47a1a672-4e8a-4965-a7d4-a0823847fc96.png)

**Comment by noahbrenner** [comment]
I'm running into the same problem with volar, though with Vue 3. For now, I was able to get it working using an old version. I installed `@volar/vue-language-server@0.40.13` and in my `tsconfig.json` I set `compilerOptions.jsx = "preserve"`. Any volar version from 1.0 on doesn't work for me.

While this does work, it's not great that we can't use the current version of volar. Since it breaks at 1.0, I'm wondering if there was an API change that ALE hasn't been updated to handle. It's just odd that there aren't any error messages, it simply doesn't provide output for any language server commands.

**Comment by theahura** [comment]
Running into the same issue. I can confirm that as of July volar is still not working -- maybe we should upgrade this to a bug if it seems to be an API breaking change?

**Reply by theahura** [reply]
@creativenull not sure if you are still keeping the volar module up to date, but figured I'd loop you in in case you have any ideas

I notice too that @KKKiraSong actually did comment on the original PR but it was moved to discussions. That said, if there's no further movement here I might open a bug since this seems to be a failure on ale's side?

**Reply by creativenull** [reply]
I don't use ALE for lsp anymore (switched to nvim-lsp). But let me have a look and see if this can be fixed as I also work with Vue on my daily job.

**Comment by creativenull** [comment]
@theahura @noahbrenner @KKKiraSong 

Apologies for the delay, but I've made a PR with the changes. (https://github.com/dense-analysis/ale/pull/4552)

Give my branch a try and see if that fixes your problem in the meantime

```vimscript
" Example with plug
Plug 'creativenull/ale', { 'branch': 'fix-volar-v1' }
```

**Reply by theahura** [reply]
thanks so much for the super fast turnaround! will give this a shot tomorrow morning :grin:

**Comment by hsanson** [comment]
Hopefully fixed with #4552
